src/checker.rs

```diff
@@ -49,7 +54,66 @@ mod tests {
         let profile = "release";
 
         let expected = format!("{}/target/{}/deps", pwd, profile);
-        assert_eq!(expected, get_base_path().unwrap());
+        let base_path = get_base_path(env::current_exe().unwrap()).unwrap();
+        assert_eq!(expected, base_path)
+    }
+
+    #[test]
+    fn test_download() -> Result<()> {
+        let base_url = generate_base_url("2.0.3");
+        let base_path = get_base_path(env::current_exe().unwrap()).unwrap();
+        let os_type = get_os_type(&sys_info::os_type()?)?;
+
+        let filename = generate_filename(&os_type, get_architecture().unwrap());
+
+        let result = download(&base_url, &base_path, &filename);
+        let tar_path = format!("{}/{}.tar.gz", base_path, filename);
+        assert!(result.is_ok());
+        assert!(true, path_exists(&tar_path));
+        assert!(fs::remove_file(&tar_path).is_ok());
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_unpack() {
+        let archive_name = "archive.tar.gz";
+        let file_name = "file.txt";
+
+        // This block ensures every opened file is closed after the scope ends
+        {
```

Comment:
More than expected. I thought of creating the archive by hand and reusing the same archive every time. But this is even better!